presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotColumnUtils.java

```diff
@@ -39,14 +51,60 @@ private PinotColumnUtils()
     {
         return pinotTableSchema.getColumnNames().stream()
                 .filter(columnName -> !columnName.startsWith("$")) // Hidden columns starts with "$", ignore them as we can't use them in PQL
-                .map(columnName -> new PinotColumn(columnName, getPrestoTypeFromPinotType(pinotTableSchema.getFieldSpecFor(columnName))))
+                .map(columnName -> new PinotColumn(columnName, getPrestoTypeFromPinotType(pinotTableSchema.getFieldSpecFor(columnName)), isNullableColumnFromPinotType(pinotTableSchema.getFieldSpecFor(columnName)),
+                        getCommentFromPinotType(pinotTableSchema.getFieldSpecFor(columnName))))
                 .collect(toImmutableList());
     }
 
+    private static String getCommentFromPinotType(FieldSpec field)
```

Comment:
I am missing how the "comment" and "nullable" flag are being used eventually ? I didn't catch that.

Reply:
nullable is used to simplify the applications(E.g. powerbi will generate a more complex query for count distinct, is this column is nullable).

Comment is more used as a reference for user to link Pinot schemas.

Reply:
Below is a sample schema
```presto:default> select * from "pinot"."information_schema"."columns" WHERE table_schema LIKE 'default' AND table_name LIKE 'meetuprsvp' ESCAPE '\';
 table_catalog | table_schema | table_name |  column_name  | ordinal_position | column_default | is_nullable | data_type |  comment  | extra_info
---------------+--------------+------------+---------------+------------------+----------------+-------------+-----------+-----------+------------
 pinot         | default      | meetuprsvp | venue_name    |                1 | NULL           | YES         | varchar   | DIMENSION | NULL
 pinot         | default      | meetuprsvp | rsvp_count    |                2 | NULL           | YES         | integer   | METRIC    | NULL
 pinot         | default      | meetuprsvp | group_city    |                3 | NULL           | YES         | varchar   | DIMENSION | NULL
 pinot         | default      | meetuprsvp | event_id      |                4 | NULL           | YES         | varchar   | DIMENSION | NULL
 pinot         | default      | meetuprsvp | group_country |                5 | NULL           | YES         | varchar   | DIMENSION | NULL
 pinot         | default      | meetuprsvp | group_id      |                6 | NULL           | YES         | bigint    | DIMENSION | NULL
 pinot         | default      | meetuprsvp | group_name    |                7 | NULL           | YES         | varchar   | DIMENSION | NULL
 pinot         | default      | meetuprsvp | event_name    |                8 | NULL           | YES         | varchar   | DIMENSION | NULL
 pinot         | default      | meetuprsvp | mtime         |                9 | NULL           | YES         | timestamp | TIME      | NULL
 pinot         | default      | meetuprsvp | event_time    |               10 | NULL           | YES         | bigint    | DIMENSION | NULL
(10 rows)

Reply:
Ah, so who consumes this "comment" field ? Is there some UI tool that consumes that or is this just for better instrumentation ?

Reply:
It's majorly consumed by users directly for now. UI could incorporate this, e.g. Superset, Thirdeye, just it requires changes also.

Reply:
Got it !. I was missing the fact that all we are doing is populating the ColumnMetadata notions of nullable and comment.